lib/sinon/sandbox.js

```diff
@@ -278,6 +281,33 @@ function Sandbox() {
         return replacement;
     };
 
+    sandbox.define = function define(object, property, value) {
+        const descriptor = getPropertyDescriptor(object, property);
+
+        if (typeof descriptor !== "undefined") {
+            throw new TypeError(
+                `Cannot define the already existing property ${valueToString(
+                    property
+                )}`
+            );
+        }
+
+        if (typeof value === "undefined") {
+            throw new TypeError("Expected value argument to be defined");
+        }
```

Comment:
It is possible to have a descriptor with a `value` of `undefined`, which is different from not being defined, but this might be fine in practice. Edge cases :)

Reply:
I think this is _fine_ 😄